.github/workflows/release-check.yml

```diff
@@ -0,0 +1,312 @@
+            scheme: ApolloTests
+            test-plan: Apollo-CITestPlan
+            name: Apollo Unit Tests - iOS 17.0
+            run-js-tests: false
+          # tvOS_current
+          - destination: platform=tvOS Simulator,OS=17.0,name=Apple TV
+            scheme: ApolloTests
+            test-plan: Apollo-CITestPlan
+            name: Apollo Unit Tests - tvOS ${{ vars.TVOS_VERSION }}
+            run-js-tests: false
+          # Codegen CLI Test
+          - destination: platform=macOS,arch=x86_64
+            scheme: CodegenCLITests
+            test-plan: CodegenCLITestPlan
+            name: Codegen CLI Unit Tests - macOS
+            run-js-tests: false
+          # CodegenLib Test
+          - destination: platform=macOS,arch=x86_64
+            scheme: ApolloCodegenTests
+            test-plan: Apollo-Codegen-CITestPlan
+            name: Codegen Lib Unit Tests - macOS
+            run-js-tests: true
+    name: ${{ matrix.name }}
+    steps:
+    - uses: maxim-lobanov/setup-xcode@v1
+      with:
+        xcode-version: ${{ env.XCODE_VERSION }}
+    - name: Checkout Repo
+      uses: actions/checkout@v3
+    - name: Setup Repo
+      id: setup-repo
+      uses: ./.github/actions/setup-repo
+    # Caching for apollo-ios and apollo-ios-codegen SPM dependencies
+    # - uses: actions/cache@v3
+    #   with:
+    #     path: ./DerivedData/SourcePackages
+    #     key: ${{ runner.os }}-spm-${{ hashFiles('./apollo-ios/Package.resolved') }}-${{ hashFiles('./apollo-ios-codegen/Package.resolved') }}
+    - name: Run Tuist Generation
+      uses: tuist/tuist-action@0.13.0
+      with:
+          command: 'generate'
+          arguments: ''
+    - name: Build and Test
+      uses: ./.github/actions/build-and-run-unit-tests
+      with:
+        destination: ${{ matrix.destination }}
+        scheme: ${{ matrix.scheme }}
+        test-plan: ${{ matrix.test-plan }}
+    - name: Run-JS-Tests
+      if: ${{ matrix.run-js-tests == true }}
+      shell: bash
+      working-directory: apollo-ios-codegen/Sources/GraphQLCompiler/JavaScript/
+      run: |
+        npm install && npm test
+    - name: Save xcodebuild logs
+      uses: actions/upload-artifact@v3
+      with:
+        name: ${{ matrix.name }}-logs
+        path: |
+          DerivedData/Logs/Build
+    - name: Save crash logs
+      uses: actions/upload-artifact@v3
+      with:
+        name: ${{ matrix.name }}-crashes
+        path: |
+          ~/Library/Logs/DiagnosticReports
+    - name: Zip Result Bundle
+      shell: bash
+      working-directory: TestResults
+      run: |
+        zip -r ResultBundle.zip ResultBundle.xcresult
+    - name: Save test results
+      uses: actions/upload-artifact@v3
+      with:
+        name: ${{ matrix.name }}-results
+        path: |
+          TestResults/ResultBundle.zip
+
+  legacy-unit-test:
+    runs-on: macos-11
+    timeout-minutes: 20
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          # iOS 14.4
```

Comment:
I wonder if the legacy tests should test on the latest of each of the major iOS versions, e.g.: `13.7`, `14.8.1`, `15.7.9`, `16.7.1`?

I kinda regard `14.4` as a special case that we could have another job for; I'm not certain if the differing behavior would get caught by `13.7`?

Reply:
Guess it depends on how fine grained we want to be on testing older versions, originally i know we talked about doing 14.4 and newest, but I'm open to adding other versions as well